docker-compose.full-demo.yml

```diff
@@ -0,0 +1,158 @@
+version: '3.2'
+
+# This docker-compose file sets up the services
+# and shared infrastructure on a common network called
+# `microservices_demo_1`. Because of this, all services
+# are able to communicate by their `container_name`.
+
+# Only the load balancer ports are mapped out. However, we
+# also leave the ports for the databases and the rabbit queue
+# mapped out so that you can connect GUI-based tooling to them,
+# or access the RabbitMQ admin interface
+
+services:
+  load_balancer:
+    image: 'nginx:1.23'
+
+    # The load balancer depends on the hostnames of the messenger application
+    # So it needs to start once those are created.
+    depends_on:
+      - messenger
+      - notifier
+    volumes:
+      - './load_balancer/nginx.conf:/etc/nginx/nginx.conf:ro'
+
+    # The load balancer is the only service that has ports exposed out.
+    ports:
+      - '80:80'
+    environment:
+      - NGINX_PORT=80
+    networks:
+      - microservices_demo_1
+
+  rabbitmq:
+    image: 'rabbitmq:3.11.4-management-alpine'
+    container_name: rabbitmq
+    hostname: microservices_march
+
+    # The healthcheck is critical, because we need the RabbitMQ
+    # server to actually be up and accepting connections before we
+    # start the applications, otherwise they will crash.
+    healthcheck:
+      test: rabbitmq-diagnostics -q ping
+      interval: 30s
+      timeout: 30s
+      retries: 3
+    ports:
+      - '5672:5672'
+      - '15672:15672'
+    volumes:
+      - 'rabbit-data:/var/lib/rabbitmq/'
+      - 'rabbit-log:/var/log/rabbitmq/'
+    networks:
+      - microservices_demo_1
+
+  messenger:
+    image: microservices_demo/messenger
+    container_name: messenger
+
+    # This uses the `Dockerfile` in the `messenger` repository
+    # to build the container for the service.
+    # If you've made changes to the `messenger` application,
+    # run `docker-compose -f docker-compose.full-demo.yml build`
+    # to make sure this file is using the updated container.
+    build:
+      context: ../messenger
```

Comment:
neat, don't think I knew about this option in docker-compose

Reply:
It's great! Also lets this quick setup eat whatever improvements we make to the real dockerfiles, and make it so there are no special cases for this contrived packaging of the demo.